tests/e2e/tests/pilot/pilot_test.go

```diff
@@ -353,24 +353,26 @@ func (t *testConfig) Teardown() (err error) {
 }
 
 func getApps() []framework.App {
-	appsWithSidecar = []string{"a-", "b-", "c-", "d-", "headless-"}
+	appsWithSidecar = []string{"a-", "b-", "c-", "d-", "e-", "f-", "headless-"}
 	return []framework.App{
 		// deploy a healthy mix of apps, with and without proxy
-		getApp("t", "t", 1, 8080, 80, 9090, 90, 7070, 70, "unversioned", false, false, false, true),
-		getApp("a", "a", 1, 8080, 80, 9090, 90, 7070, 70, "v1", true, false, true, true),
-		getApp("b", "b", 1, 80, 8080, 90, 9090, 70, 7070, "unversioned", true, false, true, true),
-		getApp("c-v1", "c", 1, 80, 8080, 90, 9090, 70, 7070, "v1", true, false, true, true),
-		getApp("c-v2", "c", 1, 80, 8080, 90, 9090, 70, 7070, "v2", true, false, true, false),
-		getApp("d", "d", 1, 80, 8080, 90, 9090, 70, 7070, "per-svc-auth", true, false, true, true),
-		getApp("headless", "headless", 1, 80, 8080, 10090, 19090, 70, 7070, "unversioned", true, true, true, true),
+		getApp("t", "t", 1, 8080, 80, 9090, 90, 7070, 70, "unversioned", false, false, false, true, false),
+		getApp("a", "a", 1, 8080, 80, 9090, 90, 7070, 70, "v1", true, false, true, true, false),
+		getApp("b", "b", 1, 80, 8080, 90, 9090, 70, 7070, "unversioned", true, false, true, true, false),
+		getApp("c-v1", "c", 1, 80, 8080, 90, 9090, 70, 7070, "v1", true, false, true, true, false),
+		getApp("c-v2", "c", 1, 80, 8080, 90, 9090, 70, 7070, "v2", true, false, true, false, false),
+		getApp("d", "d", 1, 80, 8080, 90, 9090, 70, 7070, "per-svc-auth", true, false, true, true, false),
+		getApp("e", "e", 1, 80, 8080, 90, 9090, 70, 7070, "v1", true, false, true, true, true),
+		getApp("f", "f", 1, 80, 8080, 90, 9090, 70, 7070, "per-svc-auth", true, false, true, true, true),
```

Comment:
Rather than adding two more pods to test this, I would simply change the existing pods so that the sniffer goes through ALL tests that we do on these pods t,a,b,c,d. Otherwise, you are only testing a subset of the functionality (reachability and not all the routing issues that we test for). The otehr advantage is that if you change the port names of these existing workloads, you dont have to add any new test. If all the existing tests pass, then you know that the sniffing logic works properly.
If you want to be truly adventurous [which you should be in this case], simply rename all the ports to arbitrary values [i.e. not starting with http2 or whatever]. Or for sake of clarity, you can change http2-two to two-http2, one-http etc. In my opinion, that is all you need to change in the e2e test. You dont have to add any special variables for sniffing, etc., nor do you have to add a new test.

Reply:
Changed the name to foo, bar.. and all pilot e2e tests passed locally.